pkg/eraser/eraser_test.go

```diff
@@ -85,12 +95,21 @@ func TestParseEndpoint(t *testing.T) {
 			endpoint: "unix://  ",
 			protocol: "",
 			addr:     "",
-			err:      ErrParsing,
+			err:      &url.Error{},
 		},
 	}
 	for _, tc := range testCases {
 		p, a, e := parseEndpoint(tc.endpoint)
-		if p != tc.protocol || a != tc.addr || !errors.Is(e, tc.err) {
+
+		as := tc.err
+		var errAux bool
+		if as == nil {
+			errAux = !errors.Is(e, as)
+		} else {
+			errAux = !errors.As(e, &as)
```

Comment:
I don't think this is doing what we wanted.
As an example: https://play.golang.org/p/a4qR3ervYOw.

There doesn't really seem to be a good way to assert the url.Error and also other errors we want to catch with the `errors` package without having the literal error that `url.Error` is wrapping (in which case we could use `errors.Is`).
Hence, my [suggestion](https://github.com/Azure/eraser/pull/31/files/5b1ed2b6f327cb77659991e5eeeb80e4de0a2c3a..a33cf0387adc0a25e947a09e215c2d9321496dcc#r681924378).